packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,263 @@
+import React, {createContext, forwardRef, ReactElement, ReactNode, useEffect, useMemo, useState} from 'react';
+import {style} from '../style' with {type: 'macro'};
+import {useDOMRef} from '@react-spectrum/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export type SelectBoxValueType = string | string[];
+
+export interface SelectBoxGroupProps extends StyleProps, SpectrumLabelableProps, HelpTextProps {
+  /**
+   * The SelectBox elements contained within the SelectBoxGroup.
+   */
+  children: ReactNode,
+  /**
+   * Handler that is called when the selection changes.
+   */
+  onSelectionChange: (val: SelectBoxValueType) => void,
+  /**
+   * The selection mode for the SelectBoxGroup.
+   * @default 'single'
+   */
+  selectionMode?: 'single' | 'multiple',
+  /**
+   * The current selected value (controlled).
+   */
+  value?: SelectBoxValueType,
+  /**
+   * The default selected value.
+   */
+  defaultValue?: SelectBoxValueType,
+  /**
+   * The size of the SelectBoxGroup.
+   * @default 'M'
+   */
+  size?: 'S' | 'M' | 'L' | 'XL',
+  /**
+   * The axis the SelectBox elements should align with.
+   * @default 'vertical'
+   */
+  orientation?: Orientation,
+  /**
+   * Whether the SelectBoxGroup should be displayed with an emphasized style.
+   */
+  isEmphasized?: boolean,
+  /**
+   * Number of columns to display the SelectBox elements in.
+   * @default 2
+   */
+  numColumns?: number,
+  /**
+   * Gap between grid items.
+   * @default 'default'
+   */
+  gutterWidth?: 'default' | 'compact' | 'spacious',
+  /**
+   * Whether the SelectBoxGroup is required.
+   */
+  isRequired?: boolean,
+  /**
+   * Whether the SelectBoxGroup is disabled.
+   */
+  isDisabled?: boolean
+}
+
+interface SelectBoxContextValue {
+  allowMultiSelect?: boolean,
+  value?: SelectBoxValueType,
+  size?: 'S' | 'M' | 'L' | 'XL',
+  orientation?: Orientation,
+  isEmphasized?: boolean
+}
+
+// Utility functions
+const unwrapValue = (value: SelectBoxValueType | undefined): string | undefined => {
+  if (Array.isArray(value)) {
+    return value[0];
+  }
+  return value;
+};
+
+const ensureArray = (value: SelectBoxValueType | undefined): string[] => {
+  if (value === undefined) {
+    return [];
+  }
+  if (Array.isArray(value)) {
+    return value;
+  }
+  return [value];
+};
+
+export const SelectBoxContext = createContext<SelectBoxContextValue>({
+  size: 'M',
+  orientation: 'vertical'
+});
+
+export const SelectBoxGroupContext = createContext<ContextValue<Partial<SelectBoxGroupProps>, DOMRefValue<HTMLDivElement>>>(null);
+
+const gridStyles = style({
+  display: 'grid',
+  gridAutoRows: '1fr',
+  gap: {
+    gutterWidth: {
+      default: 16,
+      compact: 8,
+      spacious: 24
+    }
+  }
+}, getAllowedOverrides());
+
+
+// Selector Group component
+interface SelectorGroupProps {
+  allowMultiSelect: boolean,
+  children: ReactNode,
+  style?: React.CSSProperties,
+  className?: string,
```

Comment:
`style` and `className` might need the `UNSAFE_` prefix. and you might want to add the `styles` prop